e3sm_diags/viewer/main.py

```diff
@@ -113,11 +115,16 @@ def insert_data_in_row(row_obj, name, url):
     return output
 
 
-def create_viewer(root_dir, parameters):
+def create_viewer(parameters: List[CoreParameter]) -> str:
     """
     Based of the parameters, find the files with the
     certain extension and create the viewer in root_dir.
     """
+    root_dir = parameters[0].results_dir
+
+    if not os.path.exists(root_dir):
+        os.makedirs(root_dir)
+
```

Comment:
Generate the `index.html` in the `results_dir` directly rather than in the `/viewer` sub-directory.